Gems/Multiplayer/Code/Source/MultiplayerSystemComponent.cpp

```diff
@@ -679,6 +662,52 @@ namespace Multiplayer
         return false;
     }
 
+    bool MultiplayerSystemComponent::HandleRequest
+    (
+        AzNetworking::IConnection* connection,
+        [[maybe_unused]] const IPacketHeader& packetHeader,
+        MultiplayerPackets::Connect& packet
+    )
+    {
+        PlayerConnectionConfig config;
+        config.m_playerConnectionId = aznumeric_cast<uint32_t>(connection->GetConnectionId());
+        config.m_playerSessionId = packet.GetTicket();
+
+        // Validate our session with the provider if any
+        ISessionHandlingProviderRequests* sessionRequests = AZ::Interface<ISessionHandlingProviderRequests>::Get();
+        if (sessionRequests != nullptr)
+        {
+            if (!sessionRequests->ValidatePlayerJoinSession(config))
+            {
+                auto visitor = [](IConnection& connection) { connection.Disconnect(DisconnectReason::TerminatedByUser, TerminationEndpoint::Local); };
+                m_networkInterface->GetConnectionSet().VisitConnections(visitor);
+                return true;
+            }
+        }
+
+        // Make sure the client that's trying to connect has the same multiplayer components
+        if (GetMultiplayerComponentRegistry()->GetSystemVersionHash() != packet.GetMultiplayerComponentVersionHash())
```

Comment:
Is this the right check? These should compare "system hash" (ours) vs "system hash" theirs right? Naming makes that unclear IMHO

Reply:
Second @lmbr-pip 's sentiment.

Reply:
good called! changed to packet.GetSystemVersionHash